src/Microsoft.DotNet.Archive/IndexedArchive.cs

```diff
@@ -0,0 +1,584 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.DotNet.Archive
+{
+    public class IndexedArchive : IDisposable
+    {
+        private class DestinationFileInfo
+        {
+            public DestinationFileInfo(string destinationPath, string hash)
+            {
+                DestinationPath = destinationPath;
+                Hash = hash;
+            }
+
+            public string DestinationPath { get; }
+            public string Hash { get; }
+        }
+
+        private class ArchiveFileInfo
+        {
+            public ArchiveFileInfo(Stream stream, string archivePath, string hash)
+            {
+                Stream = stream;
+                ArchivePath = archivePath;
+                Hash = hash;
+            }
+
+            public Stream Stream { get; set; }
+            public string ArchivePath { get; }
+            public string Hash { get; }
+            public string FileName { get { return Path.GetFileNameWithoutExtension(ArchivePath); } }
+            public string Extension { get { return Path.GetExtension(ArchivePath); } }
+
+            public long Size { get { return Stream.Length; } }
+
+        }
+
+        static string[] ZipExtensions = new[] { ".zip", ".nupkg" };
+        static string IndexFileName = "index.txt";
+
+        // maps file hash to archve path
+        // $ prefix indicates that the file is not in the archive and path is a hash
+        Dictionary<string, ArchiveFileInfo> archiveFiles = new Dictionary<string, ArchiveFileInfo>();
+        // maps file hash to external path
+        Dictionary<string, string> externalFiles = new Dictionary<string, string>();
+        // lists all extracted files & hashes
+        List<DestinationFileInfo> destFiles = new List<DestinationFileInfo>();
+        bool disposed = false;
+        ThreadLocal<SHA256> sha = new ThreadLocal<SHA256>(() => SHA256.Create());
+
+        public IndexedArchive()
+        {
+
+        }
+
+
+        private static Stream CreateTemporaryStream()
+        {
+            // return new MemoryStream();
```

Comment:
Commented out code like this should be removed. It serves no purpose.

same for the instances below.